freqtrade/exchange/exchange.py

```diff
@@ -1684,11 +1704,33 @@ def _set_leverage(
         except ccxt.BaseError as e:
             raise OperationalException(e) from e
 
+    def funding_fee_cutoff(self, d: datetime):
+        '''
+            :param d: The open date for a trade
+            :return: The cutoff open time for when a funding fee is charged
+        '''
+        return d.minute > 0 or d.second > 0
+
+    def _get_funding_fee_dates(self, d1: datetime, d2: datetime):
+        if self.funding_fee_cutoff(d1):
+            d1 += timedelta(hours=1)
+        d1 = datetime(d1.year, d1.month, d1.day, d1.hour, tzinfo=timezone.utc)
+        d2 = datetime(d2.year, d2.month, d2.day, d2.hour, tzinfo=timezone.utc)
+
+        results = []
+        d3 = d1
+        while d3 <= d2:
+            if d3.hour in self.funding_fee_times:
+                results.append(d3)
+            d3 += timedelta(hours=1)
+
+        return results
```

Comment:
I think this method would benefit from clearer variable names.

`d1` / `d2` is not very clear (maybe use `d_start`, `d_end`?) - and `d3` seems to be unnecessary it's an alias of `d1` - which is never used beyond the aliasing.